x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/LegacyGeoShapeWithDocValuesIT.java

```diff
@@ -1,109 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-package org.elasticsearch.xpack.spatial.search;
-
-import org.apache.lucene.tests.util.LuceneTestCase;
-import org.elasticsearch.core.UpdateForV9;
-import org.elasticsearch.geometry.Circle;
-import org.elasticsearch.index.IndexVersion;
-import org.elasticsearch.index.IndexVersions;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.search.geo.GeoShapeIntegTestCase;
-import org.elasticsearch.test.index.IndexVersionUtils;
-import org.elasticsearch.xcontent.ToXContent;
-import org.elasticsearch.xcontent.XContentBuilder;
-import org.elasticsearch.xcontent.XContentType;
-import org.elasticsearch.xpack.spatial.LocalStateSpatialPlugin;
-
-import java.io.IOException;
-import java.util.Collection;
-import java.util.Collections;
-
-import static org.elasticsearch.index.query.QueryBuilders.geoShapeQuery;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
-import static org.hamcrest.Matchers.containsString;
-
-@UpdateForV9(owner = UpdateForV9.Owner.SEARCH_ANALYTICS)
-@LuceneTestCase.AwaitsFix(bugUrl = "this is testing legacy functionality so can likely be removed in 9.0")
-public class LegacyGeoShapeWithDocValuesIT extends GeoShapeIntegTestCase {
-
-    @Override
-    protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singleton(LocalStateSpatialPlugin.class);
-    }
-
-    @Override
-    protected void getGeoShapeMapping(XContentBuilder b) throws IOException {
-        b.field("type", "geo_shape");
-        b.field("strategy", "recursive");
-    }
-
-    @Override
-    protected IndexVersion randomSupportedVersion() {
-        // legacy shapes can only be created in version lower than 8.x
-        return IndexVersionUtils.randomPreviousCompatibleVersion(random(), IndexVersions.V_8_0_0);
-    }
-
-    @Override
-    protected boolean allowExpensiveQueries() {
-        return false;
-    }
-
-    public void testMappingUpdate() {
```

Comment:
I think this test covers what we have here for legacy shapes, at least its checking the same exception is thrown

Reply:
https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapperTests.java#L376